pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1365,7 +1365,20 @@ func (c *Controller) getProxyServiceInstancesByPod(pod *v1.Pod,
 func (c *Controller) GetProxyWorkloadLabels(proxy *model.Proxy) labels.Instance {
 	pod := c.pods.getPodByProxy(proxy)
 	if pod != nil {
-		return pod.Labels
+		if _, exist := pod.Labels[model.LocalityLabel]; exist {
+			return pod.Labels
+		}
+		locality := c.getPodLocality(pod)
+		if locality == "" {
+			return pod.Labels
+		}
+		out := make(map[string]string, len(pod.Labels)+1)
+		for k, v := range pod.Labels {
+			out[k] = v
+		}
+		// Add locality labels
```

Comment:
Can you comment *why* we do it, rather than what? Its not clear why we need to do it here